### DISCUSSION TITLE
Get file before and after commit

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#34874] [Actions] [answered]
Is there a way for a workflow that is executed `on: push` to get the file **before** and **after** the commit(s) that were pushed?

**Comment by airtower-luna** [comment, marked as answer]
The [`push` event payload](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push) contains `before` and `after` fields that give you the relevant commit IDs. You can use checkout, diff, or whatever else you need with those.

One catch is that by default [`actions/checkout`](https://github.com/actions/checkout) retrieves only the latest commit. You'll need to either set `fetch-depth: 0` for the checkout step, or add a fetch command to retrieve the required parts of the repository history.
